Issue running with shared credentials assuming a role with MFA set

```
cat ~/.aws/credentials
[test]
aws_access_key_id = XXXXXXXXXXXX
aws_secret_access_key = XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
```
```
cat ~/.aws/config
[profile testing]
mfa_serial = arn:aws:iam::012345678910:mfa/enrique.garbi@example.com
source_profile = test
role_arn = arn:aws:iam::012345678910:role/admin
```
```
regions:
- eu-west-2

account-blacklist:
- "99999999999" # production

resource-types:
  targets:
  - EC2Instance

accounts:
  "012345678910": {} # testing
```
```
./aws-nuke -v --profile testing -c config.yaml
panic: assignment to entry in nil map

goroutine 1 [running]:
github.com/rebuy-de/aws-nuke/pkg/awsutil.(*Credentials).Session(0xc4201b4460, 0x3115af8, 0x9, 0xffffffffffffffff, 0xc42019f980, 0x10ce901)
	/go/src/github.com/rebuy-de/aws-nuke/pkg/awsutil/session.go:111 +0x10b
github.com/rebuy-de/aws-nuke/pkg/awsutil.NewAccount(0x7ffeefbffb70, 0x7, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc4201615f0, ...)
	/go/src/github.com/rebuy-de/aws-nuke/pkg/awsutil/account.go:20 +0xee
github.com/rebuy-de/aws-nuke/cmd.NewRootCommand.func2(0xc420098d80, 0xc42009af00, 0x0, 0x5, 0x0, 0x0)
	/go/src/github.com/rebuy-de/aws-nuke/cmd/root.go:49 +0xcf
github.com/rebuy-de/aws-nuke/vendor/github.com/spf13/cobra.(*Command).execute(0xc420098d80, 0xc420070070, 0x5, 0x5, 0xc420098d80, 0xc420070070)
	/go/src/github.com/rebuy-de/aws-nuke/vendor/github.com/spf13/cobra/command.go:643 +0x3ef
github.com/rebuy-de/aws-nuke/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc420098d80, 0xc4201a06d8, 0xc42009ab90, 0xc42009abe0)
	/go/src/github.com/rebuy-de/aws-nuke/vendor/github.com/spf13/cobra/command.go:734 +0x339
github.com/rebuy-de/aws-nuke/vendor/github.com/spf13/cobra.(*Command).Execute(0xc420098d80, 0xc4200001a0, 0xc4200001a0)
	/go/src/github.com/rebuy-de/aws-nuke/vendor/github.com/spf13/cobra/command.go:692 +0x2b
main.main()
	/go/src/github.com/rebuy-de/aws-nuke/main.go:21 +0x2f
```

--- a/cmd/nuke.go
+++ b/cmd/nuke.go
@@ -129,7 +129,7 @@ func (n *Nuke) Scan() error {
 	queue := make(Queue, 0)
 
 	for _, regionName := range n.Config.Regions {
-		sess, err := n.Account.Session(regionName)
+		sess, err := n.Account.NewSession(regionName)
 		if err != nil {
 			return err
 		}

--- a/pkg/awsutil/account.go
+++ b/pkg/awsutil/account.go
@@ -17,7 +17,7 @@ func NewAccount(creds Credentials) (*Account, error) {
 		Credentials: creds,
 	}
 
-	defaultSession, err := account.Session(DefaultRegionID)
+	defaultSession, err := account.NewSession(DefaultRegionID)
 	if err != nil {
 		return nil, err
 	}
@@ -27,7 +27,7 @@ func NewAccount(creds Credentials) (*Account, error) {
 		return nil, err
 	}
 
-	globalSession, err := account.Session(GlobalRegionID)
+	globalSession, err := account.NewSession(GlobalRegionID)
 	if err != nil {
 		return nil, err
 	}

--- a/pkg/awsutil/session.go
+++ b/pkg/awsutil/session.go
@@ -101,6 +101,10 @@ func (c *Credentials) NewSession(region string) (*session.Session, error) {
 }
 
 func (c *Credentials) Session(region string) (*session.Session, error) {
+	if c.cache == nil {
+		c.cache = map[string]*session.Session{}
+	}
+
 	sess, ok := c.cache[region]
 	if ok {
 		return sess, nil

--- a/pkg/awsutil/session.go
+++ b/pkg/awsutil/session.go
@@ -24,8 +24,6 @@ type Credentials struct {
 	AccessKeyID     string
 	SecretAccessKey string
 	SessionToken    string
-
-	cache map[string]*session.Session
 }
 
 func (c *Credentials) HasProfile() bool {
@@ -98,24 +96,6 @@ func (c *Credentials) NewSession(region string) (*session.Session, error) {
 	sess.Handlers.Validate.PushFront(skipGlobalHandler(global))
 
 	return sess, nil
-}
-
-func (c *Credentials) Session(region string) (*session.Session, error) {
-	if c.cache == nil {
-		c.cache = map[string]*session.Session{}
-	}
-
-	sess, ok := c.cache[region]
-	if ok {
-		return sess, nil
-	}
-
-	sess, err := c.NewSession(region)
-	if err != nil {
-		c.cache[region] = sess
-	}
-
-	return sess, err
 }
 
 func skipMissingServiceInRegionHandler(r *request.Request) {

